numpyro/infer/elbo.py

```diff
@@ -311,53 +312,115 @@ class RenyiELBO(ELBO):
+                size = model_trace[frame.name]["args"][0]
+                if size > subsample_size and frame.name not in indep_subsample_plates:
+                    common_plates = common_plates - {frame}
+                else:
+                    common_plate_scale = common_plate_scale * size / subsample_size
+            common_plate_dims = {frame.dim for frame in common_plates}
+
+            log_densities = {}
+            for trace_type, tr in {"guide": guide_trace, "model": model_trace}.items():
+                log_densities[trace_type] = 1.0
+                for site in tr.values():
+                    if site["type"] != "sample":
+                        continue
+                    log_prob = site["log_prob"]
+                    squeeze_axes = ()
+                    for dim in range(log_prob.ndim):
+                        neg_dim = dim - log_prob.ndim
+                        if neg_dim in common_plate_dims:
+                            continue
+                        log_prob = jnp.sum(log_prob, axis=dim, keepdims=True)
+                        squeeze_axes = squeeze_axes + (dim,)
+                    log_prob = jnp.squeeze(log_prob, squeeze_axes)
+                    log_densities[trace_type] = log_densities[trace_type] + log_prob
+
             # log p(z) - log q(z)
-            elbo = model_log_density - guide_log_density
-            return elbo
+            elbo = log_densities["model"] - log_densities["guide"]
+            # Do not scale elbo at common plate dimensions.
+            return elbo / common_plate_scale, common_plate_scale
+
+        if self.create_plates is not None:
+            subkey, rng_key = random.split(rng_key)
+            plates = self._create_plates(subkey, *args, **kwargs)
+            model = substitute(model, data=plates)
+            guide = substitute(guide, data=plates)
+            plate_names = tuple(plates.keys())
+            compute_elbo_fn = partial(
+                single_particle_elbo, indep_subsample_plates=plate_names
+            )
+        else:
+            compute_elbo_fn = single_particle_elbo
 
         rng_keys = random.split(rng_key, self.num_particles)
-        elbos = vmap(single_particle_elbo)(rng_keys)
+        elbos, common_plate_scale = vmap(compute_elbo_fn)(rng_keys)
         scaled_elbos = (1.0 - self.alpha) * elbos
-        avg_log_exp = logsumexp(scaled_elbos) - jnp.log(self.num_particles)
+        avg_log_exp = logsumexp(scaled_elbos, axis=0) - jnp.log(self.num_particles)
         weights = jnp.exp(scaled_elbos - avg_log_exp)
         renyi_elbo = avg_log_exp / (1.0 - self.alpha)
-        weighted_elbo = jnp.dot(stop_gradient(weights), elbos) / self.num_particles
-        return -(stop_gradient(renyi_elbo - weighted_elbo) + weighted_elbo)
+        weighted_elbo = (stop_gradient(weights) * elbos).mean(0)
+        loss = -(stop_gradient(renyi_elbo - weighted_elbo) + weighted_elbo)
+        return loss.sum() * jnp.mean(common_plate_scale)
```

Comment:
is this a real mean or could it be `common_plate_scale[0]`?

Reply:
it could be common_plate_scale[0].